src/services/utxoSumForAddress.ts

```diff
@@ -1,25 +1,52 @@
 import {errMsgs, UtilEither} from "../utils";
 import {Pool} from "pg";
+import { UtxoSumResponse } from "../Transactions/types";
 
-export const askUtxoSumForAddresses = async (pool: Pool, addresses: string[]): Promise<UtilEither<string>> => {
+export const askUtxoSumForAddresses = async (pool: Pool, addresses: string[]): Promise<UtilEither<UtxoSumResponse>> => {
   // TODO: support for payment keys
-    const sqlQuery = `
-    SELECT "public"."utxo_view"."value" AS "value"
-    FROM "public"."utxo_view"
-    WHERE "public"."utxo_view"."address" = any(($1)::varchar array)
-    GROUP BY "public"."utxo_view"."value"
-    ORDER BY "public"."utxo_view"."value" ASC
-  `;
+    const adaQuery = `
+      SELECT "public"."utxo_view"."value" AS "value"
+      FROM "public"."utxo_view"
+      WHERE "public"."utxo_view"."address" = any(($1)::varchar array)
+      GROUP BY "public"."utxo_view"."value"
+      ORDER BY "public"."utxo_view"."value" ASC
+    `;
+
+    const tokensQuery = `
+      SELECT SUM(ma_utxo.quantity) amount,
+        encode(ma_utxo.policy, 'hex') as policy,
+        convert_from(ma_utxo.name, 'UTF8') as name
```

Comment:
Not sure if the `name` is encoded as utf8. It seems to me this is actually interpreted as ASCII code in Yoroi (https://github.com/Emurgo/yoroi-frontend/blob/develop/packages/yoroi-extension/app/stores/stateless/tokenHelpers.js#L29). I'd suggest just return the hex string. Please correct me @vsubhuman .